deepchem/feat/molecule_featurizers/mat_featurizer.py

```diff
@@ -55,24 +97,115 @@ def atom_features(self, atom: RDKitAtom) -> np.ndarray:
     ----------
     Atom_features: ndarray
       Numpy array containing atom features.
-
     """
     attrib = []
     attrib += one_hot_encode(atom.GetAtomicNum(),
                              [5, 6, 7, 8, 9, 15, 16, 17, 35, 53, 999])
     attrib += one_hot_encode(len(atom.GetNeighbors()), [0, 1, 2, 3, 4, 5])
     attrib += one_hot_encode(atom.GetTotalNumHs(), [0, 1, 2, 3, 4])
 
-    if self.one_hot_formal_charge:
-      attrib += one_hot_encode(atom.GetFormalCharge(), [-1, 0, 1])
-    else:
-      attrib.append(atom.GetFormalCharge())
+    attrib += one_hot_encode(atom.GetFormalCharge(),
+                             [-5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5])
 
     attrib.append(atom.IsInRing())
     attrib.append(atom.GetIsAromatic())
 
     return np.array(attrib, dtype=np.float32)
 
+  def construct_node_features_matrix(self, mol: RDKitMol) -> np.ndarray:
+    """
+    This function constructs a matrix of atom features for all atoms in a given molecule using the atom_features function.
+
+    Parameters
+    ----------
+    mol: RDKitMol
+      RDKit Mol object.
+
+    Returns
+    ----------
+    Atom_features: ndarray
+      Numpy array containing atom features.
+    """
+    return np.array([self.atom_features(atom) for atom in mol.GetAtoms()])
+
+  def add_dummy_node(
```

Comment:
Should this be a private method? Not sure if any DeepChem users would want to directly call this method